core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -49,7 +54,7 @@
      * @param filterBuilder The query to wrap in a constant score query
      */
     public ConstantScoreQueryBuilder(QueryBuilder filterBuilder) {
-        this.filterBuilder = filterBuilder;
+        this.filterBuilder = Objects.requireNonNull(filterBuilder);
```

Comment:
it is good to check for null values, but the default is null, which means that if remove null checks (e.g. in doXContent) we might have NPEs coming from java api at least. I am on the fence on what should be the right behaviour, here too we validate that the field is present in the json, but we don't validate that the field was set through java api. We should unify our behaviour through the different queries and do the same in both parser and builder, as much as possible.